File: apps/mobile/ios/RoamPulseWidget/Shared/WidgetUiState.swift
import Foundation

/// Everything the widget renders, already formatted — the output of
/// `mapToWidgetUiState`. Mirrors Android's `WidgetUiState`/`ConnectionTone`
/// (apps/mobile/android/.../widget/WidgetUiState.kt) field-for-field, so
/// both platforms render the same information from the same contract,
/// even though neither shares code with the other.
struct WidgetUiState {
    let connectionLabel: String
    let connectionTone: ConnectionTone
    let carrierLine: String?
    let destinationLine: String?
    let dataRemainingLabel: String?
    let dataPercentRemaining: Int?
    let expiryLabel: String?
    let lastUpdatedLabel: String?
}

enum ConnectionTone {
    case positive
    case warning
    case negative
    case neutral
}

File: apps/mobile/android/app/src/main/kotlin/com/example/roam_pulse/widget/WidgetUiState.kt
package com.example.roam_pulse.widget

/**
 * Everything the widget renders, already formatted — the output of
 * [mapToWidgetUiState]. Kept free of Android framework types so it (and
 * the pure function that produces it) can be unit-tested with plain
 * JUnit, no Robolectric/instrumentation — the same pure-core pattern as
 * the Go diagnostic engine and Dart's Chaos Mode decision function.
 */
data class WidgetUiState(
    val connectionLabel: String,
    val connectionTone: ConnectionTone,
    val carrierLine: String?,
    val destinationLine: String?,
    val dataRemainingLabel: String?,
    val dataPercentRemaining: Int?,
    val expiryLabel: String?,
    val lastUpdatedLabel: String?,
)

enum class ConnectionTone { POSITIVE, WARNING, NEGATIVE, NEUTRAL }
